feat: 유튜브 인기급상승 영상 수집 슬랙봇 구현

potatochips178 opened the pull request:
## 📋 변경사항 요약

유튜브 인기급상승 영상 수집 기능을 **슬랙봇**으로 구현하기 위한 완전한 PRD 문서와 구현 작업 목록을 추가했습니다.

### 🎯 주요 기능

#### 1. PRD 문서 작성
- **파일**: `docs/prd/active/prd-youtube-trending-collector.md`
- **목표**: 마케팅 담당자를 위한 슬랙봇 기반 트렌드 분석 시스템
- **핵심 기능**: 매일 오전 11시 자동 데이터 수집, 슬랙 명령어를 통한 조회

#### 2. 슬랙 명령어 시스템
- `/trending` - 오늘의 인기급상승 영상 조회
- `/trending-channel [채널명]` - 특정 채널의 인기급상승 이력 조회
- `/trending-stats` - 주간/월간 트렌드 통계 조회
- `/trending-history [날짜]` - 특정 날짜의 순위 조회
- `/trending-alert [채널명]` - 특정 채널 알림 설정

#### 3. 자동화 기능
- **매일 오전 11시**: 유튜브 데이터 수집 (상위 20개 영상, 쇼츠 제외)
- **매일 오전 11시 30분**: 슬랙 채널로 자동 트렌드 리포트 전송
- **실시간 알림**: 설정된 채널이 인기급상승 진입 시 즉시 알림

#### 4. 작업 목록 생성
- **파일**: `docs/tasks/tasks-prd-youtube-trending-collector.md`
- **총 45개 하위 작업**으로 구성된 상세한 구현 계획
- **6개 주요 영역**: 데이터 모델, 수집 시스템, 스케줄링, 슬랙봇 명령어, 메시지 포맷팅, 배포

### 🔧 기술 스택

- **백엔드**: Spring Boot + JPA + MySQL
- **슬랙 통합**: Slack SDK for Java, Slash Commands, Block Kit
- **데이터 수집**: 웹 스크래핑 (Selenium/Jsoup)
- **스케줄링**: Spring Boot @Scheduled
- **메시지 포맷**: 슬랙 블록 킷 활용

### 📊 성공 지표

- 데이터 수집 성공률: 95% 이상
- 슬랙봇 응답 시간: 3초 이내
- 슬랙 명령어 성공률: 98% 이상
- 일일 리포트 전송 성공률: 100%

### 🗂️ 파일 구조

```
docs/
├── prd/active/
│   └── prd-youtube-trending-collector.md    # 슬랙봇 PRD 문서
├── rules/
│   └── generate-tasks.md                    # 파일명 수정
└── tasks/
    └── tasks-prd-youtube-trending-collector.md  # 45개 작업 목록
```

### 📝 커밋 히스토리

- `feat: 유튜브 인기급상승 영상 수집 기능 PRD 및 작업 목록 추가`
- `refactor: 슬랙봇 구현으로 변경`

### 🔄 다음 단계

1. PRD 및 작업 목록 최종 검토
2. 슬랙 앱 생성 및 권한 설정
3. 개발 환경 구성
4. 1.0 데이터 모델 및 데이터베이스 설계 착수

---

**관련 이슈**: 유튜브 트렌드 분석 시스템 구축  
**예상 개발 기간**: 3-4주  
**담당자**: 개발팀

potatochips178 committed:
feat: 유튜브 인기급상승 영상 수집 기능 PRD 및 작업 목록 추가

- 유튜브 트렌딩 데이터 수집 기능에 대한 상세한 PRD 문서 작성
- 마케팅 담당자를 위한 트렌드 분석 시스템 요구사항 정의
- PRD 기반 구현 작업 목록 생성 (35개 하위 작업)
- 데이터 수집, 스케줄링, 시각화, 분석 기능 포함
- generate-tasks.md 파일명 오타 수정

potatochips178 committed:
refactor: 슬랙봇 구현으로 변경

- PRD 문서를 슬랙봇 구현에 맞게 전면 수정
- 웹 인터페이스 대신 슬랙 명령어 중심으로 변경
- 5개 주요 슬랙 명령어 정의
- 슬랙 블록 킷 활용한 메시지 포맷팅 요구사항 추가
- 작업 목록을 슬랙봇 구현에 맞게 재구성

soo5717 commented on docs/prd/active/prd-youtube-trending-collector.md:
```diff
@@ -0,0 +1,143 @@
+
+- **데이터베이스:** PostgreSQL 또는 MySQL 사용 권장
+- **스케줄링:** Cron job 또는 Spring Boot의 @Scheduled 어노테이션 활용
+- **웹 스크래핑:** Selenium 또는 Jsoup 라이브러리 사용
```
google API 사용하도록 변경해주세요

soo5717 commented on docs/prd/active/prd-youtube-trending-collector.md:
```diff
@@ -0,0 +1,143 @@
+- **웹 스크래핑:** Selenium 또는 Jsoup 라이브러리 사용
+- **슬랙 API:** Slack Web API 및 Slack Bolt Framework 활용
+- **슬랙 명령어:** Slash Commands 구현
+- **슬랙 메시지:** Block Kit을 활용한 리치 메시지 포맷
```
Block Kit은 뭘까요?

potatochips178 replied:
슬랙 프레임워크 https://api.slack.com/block-kit

soo5717 commented on docs/prd/active/prd-youtube-trending-collector.md:
```diff
@@ -0,0 +1,143 @@
+- **슬랙 API:** Slack Web API 및 Slack Bolt Framework 활용
+- **슬랙 명령어:** Slash Commands 구현
+- **슬랙 메시지:** Block Kit을 활용한 리치 메시지 포맷
+- **슬랙 알림:** Incoming Webhooks 또는 Chat API 사용
```
흠.. 이것도 변경 필요할 것 같은데
어떤 방식으로 슬랙 알림 발송할지 추가 검토 부탁드려요.

soo5717 commented on docs/prd/active/prd-youtube-trending-collector.md:
```diff
@@ -0,0 +1,143 @@
+7. 슬랙봇 다운타임 시 대체 알림 방법은?
+8. 데이터 백업 및 복구 전략은?
+9. 성능 최적화를 위한 데이터 아카이빙 정책은?
+10. 슬랙 API 요청 제한 대응 방안은?
```
미해결 질문 해소하고, 
해당 섹션은 제거 후 업로드 해주세요.

potatochips178 committed:
feat: Update PRD to use YouTube Data API instead of web scraping

- Change data collection method from web scraping to YouTube Data API v3
- Update filtering logic to use duration info from API response
- Add YouTube API quota management and error handling
- Update security considerations for API key management
- Add specific error handling for API quota and authentication failures
- Version bump to 1.2

soo5717 reviewed (approved):
👍
